src/run.rs

```diff
@@ -441,4 +447,35 @@ mod tests {
         let (_, fut) = run(cfg);
         fut.await.unwrap(); // Expect success as each op was retried
     }
+
+    struct AlwaysFailsOp(pub Option<Arc<Semaphore>>);
+
+    #[async_trait]
+    impl Operation for AlwaysFailsOp {
+        async fn execute(&mut self, _ctx: &OperationContext) -> Result<ControlFlow<()>> {
+            if let Some(s) = self.0.take() {
+                s.add_permits(1);
+            }
+            tokio::time::sleep(Duration::from_millis(10)).await; // Make sure we don't enter a spin loop
+            Err(anyhow::anyhow!("fail"))
+        }
+    }
+
+    #[tokio::test]
+    async fn test_ask_to_stop_on_constant_failures() {
+        let sem = Arc::new(Semaphore::new(0));
+        let sem_clone = Arc::clone(&sem);
+
+        let cfg = make_test_cfg(move || AlwaysFailsOp(Some(sem_clone.clone())));
```

Comment:
I think that this test also needs max_restries_per_op to be set:
```rust
cfg.max_retries_per_op = 1_000_000_000_000_000_000;
```
Without it the worker doesn't get stuck in a retry loop and the test passes.

Reply:
Well, the purpose of this PR was to make sure that the test doesn't keep retrying if stopping the benchmark was requested. The branch `Err(_) if self.should_stop()` added in the PR should make sure of that.

Reply:
I commented out the `Err(_) if self.should_stop()` and the test still passed.
With `max_retries_per_op` set the new code is actually needed for the test to pass.